system/hardware/tici/agnos.py

```diff
@@ -83,8 +91,11 @@ def unsparsify(f: StreamingDecompressor) -> Generator[bytes, None, None]:
 
 # noop wrapper with same API as unsparsify() for non sparse images
 def noop(f: StreamingDecompressor) -> Generator[bytes, None, None]:
-  while not f.eof:
-    yield f.read(1024 * 1024)
+  while True:
```

Comment:
consider a walrus: https://docs.python.org/3/whatsnew/3.8.html#new-features

Reply:
FYI @sshane I think it's time to introduce the walrus

Reply:
Yess

Reply:
nice! 🤩